docs/tutorials/web/ep-webgpu.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Using WebGPU
+description: Using WebGPU
+parent: Web
+grand_parent: Tutorials
+has_children: false
+nav_order: 2
+---
+{::options toc_levels="2..4" /}
+
+# Using WebGPU Execution Provider
+{: .no_toc }
+
+This document explains how to use the WebGPU execution provider in ONNX Runtime.
+
+## Contents
+{: .no_toc}
+
+* TOC
+{:toc}
+
+
+## Basics
+
+### What is WebGPU? Should I use it?
+
+WebGPU is a new web standard for general purpose GPU compute and graphics. It is designed to be a low-level API, similar to Vulkan and Metal, and is designed to be used in the browser. It is designed to be more efficient and performant than WebGL, and is designed to be used for machine learning, graphics, and other compute tasks.
+
+WebGPU is available out-of-box in latest versions of Chrome and Edge on Windows, macOS and Android. It is also available in Firefox under a flag and Safari Technology Preview. Check [WebGPU status](https://webgpu.io/status/) for the latest information.
+
+If you are using ONNX Runtime Web for inferencing very lightweight models in you web application, and you want to have a small binary size, you can keep using the default WebAssembly (WASM) execution provider. If you want to run more complex models, or you want to take advantage of the GPU in the client's device, you can use the WebGPU execution provider.
+
+### How to use WebGPU EP in ONNX Runtime Web
+
+This section assumes you have already set up your web application with ONNX Runtime Web. If you haven't, you can follow the [Get Started](../../get-started/with-javascript/web.md) for some basic info.
+
+To use WebGPU EP, you just need to make 2 small changes:
+  1. Update your import statement:
+
+     - For HTML script tag, change `ort.min.js` to `ort.webgpu.min.js`:
+       ```html
+       <script src="https://example.com/path/ort.webgpu.min.js"></script>
+       ```
+     - For JavaScript import statement, change `onnxruntime-web` to `onnxruntime-web/webgpu`:
+       ```js
+       import * as ort from 'onnxruntime-web/webgpu';
+       ```
+
+     See [Conditional Importing](https://github.com/microsoft/onnxruntime-inference-examples/tree/main/js/importing_onnxruntime-web#conditional-importing) for details.
+
+  2. Specify 'webgpu' EP explicitly in session options:
+     ```js
+     const session = await ort.InferenceSession.create(modelPath, { ..., executionProviders: ['webgpu'] });
+     ```
+
+It is also recommended to install the latest nightly build version of ONNX Runtime Web (onnxruntime-web@dev) to get the latest features and bug fixes.
+
+## WebGPU EP features
```

Comment:
I think you are referring to https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.WebGpuExecutionProviderOption.html#preferredLayout.

I have concern on introducing this option because I am not confidence on the current performance of NCHW and I think this is still an experimental/diagnostic option.